serde_annotate_derive/src/expand.rs

```diff
@@ -4,12 +4,66 @@ use proc_macro2::TokenStream;
 use quote::quote;
 use syn::{DeriveInput, Index, Member, Result};
 
-pub fn derive(node: &DeriveInput) -> Result<TokenStream> {
-    let input = Input::from_syn(node)?;
+pub fn derive(mut node: DeriveInput) -> Result<TokenStream> {
+    let input = Input::from_syn(&node)?;
 
-    Ok(match input {
+    let annotate_imp = match input {
         Input::Struct(input) => impl_struct(input),
         Input::Enum(input) => impl_enum(input),
+    };
+
+    // In addition to the `Annotate` implementation, we also want to generate a
+    // `serde::Serialize` implementation.
+    //
+    // We use `serde`'s `remote` attribute to put the derive implementation on a helper
+    // type so we can invoke at will.
```

Comment:
Why do we need to create the helper?

Reply:
Oh, I think I see: we create the helper to hold the normal serde-derived `Serialize` impl and then implement our own `Serialize` to attempt to invoke the `AnnotatedSerializer`.